### PR TITLE
Pokedex Documentation

### PR DESCRIPTION
**Pull request by Meowsy**
Document various things relating to the Pokedex.

### CODE REVIEW

**Comment by Meowsy** [comment]
Done.

File: src/pokedex.c [hunk 1]
```diff
@@ -1898,6 +1889,9 @@ void sub_80BC890(void)
 void sub_80BC8D4(u8 dexMode, u8 sortMode)
 {
     u16 vars[3]; //I have no idea why three regular variables are stored in an array, but whatever.
+#define temp_dexCount       vars[0]
+#define temp_notNationalDex vars[1]
```
Comment (huderlem) [inline src/pokedex.c]: `temp_isHoennDex` instead of `temp_notNationalDex`

File: src/credits.c [hunk 2]
```diff
@@ -2259,7 +2259,7 @@ static u8 sub_8177224(u16 species, s16 x, s16 y, u16 position)
     u8 spriteId;
     u8 spriteId2;
 
-    spriteId = sub_80C0E9C(species, x, y, position);
+    spriteId = CreateMonSpriteFromNationalDexNumber(species, x, y, position);
```
Comment (huderlem) [inline src/credits.c]: The first arg here should be named `nationDexNum`, not `species`

File: src/pokedex.c [hunk 3]
```diff
@@ -2963,7 +2957,7 @@ void sub_80BEA24(u8 taskId)
         case 5:
             if (gTasks[taskId].data[1] == 0)
             {
-                gTasks[taskId].data[4] = (u16)sub_80C0E9C(sPokedexListItem->dexNum, 0x30, 0x38, 0);
+                gTasks[taskId].data[4] = (u16)CreateMonSpriteFromNationalDexNumber(sPokedexListItem->dexNum, 0x30, 0x38, 0);
```
Comment (huderlem) [inline src/pokedex.c]: Don't use hexadecimal format for `x` and `y` args.

File: src/pokedex.c [hunk 4]
```diff
@@ -3790,7 +3784,7 @@ void sub_80BFE38(u8 taskId)
             gTasks[taskId].data[0]++;
             break;
         case 4:
-            spriteId = sub_80C0E9C(dexNum, 0x30, 0x38, 0);
+            spriteId = CreateMonSpriteFromNationalDexNumber(dexNum, 0x30, 0x38, 0);
```
Comment (huderlem) [inline src/pokedex.c]: Don't use hex here, too

File: src/credits.c (hunk 2 above)
Comment (Meowsy) [reply src/credits.c]: Renamed to `nationalDexNum`.

**Pull request merged**